hs_dataverse/tests.py

```diff
@@ -0,0 +1,393 @@
+            'jerry@gmail.com',
+            username='jerry',
+            first_name='not a cat',
+            last_name='last_name_dog',
+            superuser=False,
+            groups=[]
+        )
+
+        metadata = [
+            {'title': {'value': 'Updated Resource Title'}},
+            {'description': {'abstract': 'Updated Resource Abstract'}},
+            {'date': {'type': 'valid', 'start_date': '1/26/2016', 'end_date': '12/31/2016'}},
+            {'date': {'type': 'created', 'start_date': '1/26/2016'}},   # will be ignored without error
+            {'date': {'type': 'modified', 'start_date': '1/26/2016'}},   # will be ignored without error
+            {'date': {'type': 'published', 'start_date': '1/26/2016'}},   # will be ignored without error
+            {'date': {'type': 'available', 'start_date': '1/26/2016'}},   # will be ignored without error
+            {'creator': {'name': 'John Smith', 'email': 'jsmith@gmail.com'}},
+            {'creator': {'name': 'Jane Smith', 'email': 'jsmith@gmail.com'}},
+            {'contributor': {'name': 'Kelvin Marshal', 'email': 'kmarshal@yahoo.com',
+                             'organization': 'Utah State University',
+                             'identifiers': {'ORCID': 'https://orcid.org/john',
+                                             'ResearchGateID': 'https://www.researchgate.net/john'}
+                             }},
+            {'coverage': {'type': 'period', 'value': {'name': 'Name for period coverage', 'start': '1/1/2000',
+                                                      'end': '12/12/2012'}}},
+            {'coverage': {'type': 'point', 'value': {'name': 'Name for point coverage', 'east': '56.45678',
+                                                     'north': '12.6789', 'units': 'decimal deg'}}},
+            {'format': {'value': 'txt/csv'}},   # will be ignored without error
+            {'format': {'value': 'zip'}},   # will be ignored without error
+            {'identifier': {'name': 'someIdentifier', 'url': "http://some.org/002"}},
+            {'language': {'code': 'fre'}},
+            {'relation': {'type': 'isPartOf', 'value': 'http://hydroshare.org/resource/001'}},
+            {'rights': {'statement': 'This is the rights statement for this resource', 'url': 'http://rights.ord/001'}},
+            {'source': {'derived_from': 'http://hydroshare.org/resource/0001'}},
+            {'subject': {'value': 'sub-1'}},
+            {'subject': {'value': 'sub-2'}}
+        ]
+
+        self.res = hydroshare.create_resource(
+            resource_type='CompositeResource',
+            owner=self.cat,
+            title='all about dog holes',
+            metadata=metadata
+        )
+
+        self.res.extra_metadata = {'name': 'John Jackson', 'email': 'jj@gmail.com'}
+        self.res.save()
+
+        self.temp_dir = tempfile.mkdtemp(prefix=self.res.short_id, suffix='_dataverse_tempdir', dir='/tmp')
+        self.temp_dir = export_bag(self)
```

Comment:
This is dumb, but it helps my bean counting.  I want your tests to come up as testing "export_bag" from utils, or as many lines of that as possible. This is a metric that gets computed when you submit a PR. So, if you can refactor export_bag so that it has an option for not contacting dataverse, publish=True, which allows me to count that I have tested it in the tests, it would be great!